infra/is-ngcc-necessary.ts

```diff
@@ -0,0 +1,58 @@
+import * as chalk from 'chalk';
+import { existsSync, readFileSync } from 'fs';
+import { resolve } from 'path';
+
+const packages = Object.keys(JSON.parse(readFileSync(resolve(__dirname, '../package.json'), 'utf8')).dependencies);
+const knownNonNgccPackages: string[] = JSON.parse(readFileSync(resolve(__dirname, './non-ngcc-packages.json'), 'utf8'));
+const knownNonNgccPackageMatchers = knownNonNgccPackages.map((pattern: string) => new RegExp(`^${pattern}$`));
+
+console.log(chalk.gray('Checking for packages that no longer need ngcc processing'));
+
+const nonNgccPackages = [];
+for(const packageName of packages) {
+  if (knownNonNgccPackageMatchers.some(matcher => matcher.test(packageName))) {
```

Comment:
Related to my earlier comment (about dropping regexes), this would become: `knownNonNgccPackages.includes(packageName)`

(Or better yet, we could convert `knownNonNgccPackages` to a `Set` for faster lookups :wink:)